packages/apps/patterns/react-ui/src/components/HaloRing/HaloRing.tsx

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { PropsWithChildren } from 'react';
+
+import { Pulse, getSize, mx } from '@dxos/react-components';
+
+export type HaloRingProps = PropsWithChildren & {
+  loading?: boolean;
+  status?: 'active' | 'inactive' | undefined;
+};
+
+export const HaloRing = (props: HaloRingProps) => {
```

Comment:
`Ring` is maybe not an ideal name for this, since it's overloaded by Tailwind’s `ring` utility classnames which are mainly used for focus states.

‘Status’ or ‘Avatar’, even ‘Frame’ or ‘Decorator’, are available and probably more suitable for what this is. Generally for things that render markup, we should keep an eye to WAI-ARIA, and [`role=status`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/status_role) is a reasonable connection to make here.